.github/workflows/crdb-api-client-npm-publish.yml

```diff
@@ -0,0 +1,110 @@
+# when the produced outputs are changed comparing to last published version.
+# 1st step (first CI run) checks that built files contain changes comparing to previous version and
+# if NPM package version incremented:
+# - if yes, it publishes to NPM registry
+# - if no, new PR created (with incremented version in package.json file) which should be merged and
+# then second step is executed (runs workflow again) and publishes package.
+name: Publish crdb-api-client package to NPM registry
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish_crdb_api_client:
+    if: github.repository == 'cockroachdb/cockroach'
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Bazel Cache
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/bazel
+        key: ${{ runner.os }}-bazel-cache
+
+    - uses: pnpm/action-setup@v2
+      with:
+        version: "8.6.10"
+
+    - name: Setup NodeJS
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16
+        registry-url: 'https://registry.npmjs.org'
+        always-auth: true
+        cache: 'pnpm'
+        cache-dependency-path: "${{ github.workspace }}/pkg/ui/pnpm-lock.yaml"
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+    - name: Build crdb-api-client package
+      run: |
+        echo "build --config=dev" >> .bazelrc.user
+        echo "build --config nolintonbuild" >> .bazelrc.user
+        echo "test --test_tmpdir=/tmp/cockroach" >> .bazelrc.user
+        ./dev doctor --interactive=false --no-cache
+        ./dev ui crdb-api-client
+
+    - name: Check for changes
+      id: check-changes
+      working-directory: pkg/ui/workspaces/crdb-api-client
+      shell: bash
+      run: |
+        if [[ $(npm diff) ]]; then
+          echo "modified=true" >> $GITHUB_OUTPUT
+        else
+          echo "modified=false" >> $GITHUB_OUTPUT
+        fi
+        PUBLISHED_VERSION=$(npm view @cockroachlabs/crdb-api-client version);
+        PACKAGE_VERSION=$(cat ./package.json | jq -r ".version");
+        echo "package_version=$PACKAGE_VERSION" >> $GITHUB_OUTPUT;
+        if [[ $PUBLISHED_VERSION == $PACKAGE_VERSION ]]; then
+          echo "same_version=true" >> $GITHUB_OUTPUT
+        else
+          echo "same_version=false" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Increment version
+      if: steps.check-changes.outputs.modified == 'true' && steps.check-changes.outputs.same_version == 'true'
+      working-directory: pkg/ui/workspaces/crdb-api-client
+      run: npm version patch
+
+    - name: Create PR to increment package version
+      if: steps.check-changes.outputs.modified == 'true' && steps.check-changes.outputs.same_version == 'true'
+      uses: peter-evans/create-pull-request@v5
+      with:
+        base: master
+        token: ${{ secrets.GH_TOKEN_PR }}
```

Comment:
You need to set `environment` to master in order for this secret to be available. See https://github.com/cockroachdb/cockroach/blob/6d90413285663dc37554be07913098e858ce26b5/.github/workflows/update_releases.yaml#L25